src/workflow-engine/charms/temporal-worker/oci_factory/notification/mattermost_notifier.py

```diff
@@ -0,0 +1,100 @@
+import json
+import logging
+import os
+import requests
+import time
+from typing import Optional
+
+
+COLORS = {
+    "success": "#33CC33",
+    "failure": "#CC0000",
+    "unknown": "#B0B0B0",
+}
+
+channel_id = os.environ.get(
+    "MATTERMOST_CHANNEL_ID", "yax1c3pmm7natd5ebw3cngyrpw"
+)  # `ROCKS Bot Test` by default
+mattermost_server = "https://chat.canonical.com"
+posts_url = f"{mattermost_server}/api/v4/posts"
+headers = {
+    "Authorization": f"Bearer {os.environ['MATTERMOST_TOKEN']}",
+    "Content-Type": "application/json",
+}
+
+logging.basicConfig(level=os.environ.get("TWC_LOG_LEVEL", "info").upper())
+
+
+def send_message(title: str, message: str) -> str:
+    payload = {
+        "channel_id": channel_id,
+        "props": {
+            "attachments": [
+                {
+                    "fallback": message,
+                    "title": title,
+                    "text": message,
+                    "color": COLORS["unknown"],
+                }
+            ]
+        },
+    }
+
+    token = os.environ.get("MATTERMOST_TOKEN", None)
+    if token is None:
+        logging.warning("Unable to notify on Mattermost, MATTERMOST_TOKEN is not set!")
+
+    data = json.dumps(payload).encode("utf-8")
+    res = requests.post(posts_url, headers=headers, data=data, timeout=10)
+    res.raise_for_status()
+    return res.json()["id"]
+
+
+# Update the message conditionally based on the success of the operation
+def update_message_cond_success(
```

Comment:
I would rename this one, you also update the message if the condition fail

Reply:
Thanks, fixed in [2674da7](https://github.com/canonical/oci-factory/commit/2674da7918c9f2eeeb71b88e1e7bf506e31498ba).